components/feature/awesomebar/src/main/java/mozilla/components/feature/awesomebar/provider/BookmarksStorageSuggestionProvider.kt

```diff
@@ -44,7 +45,8 @@ class BookmarksStorageSuggestionProvider(
             AwesomeBar.Suggestion(
                 provider = this@BookmarksStorageSuggestionProvider,
                 id = it.guid,
-                icon = icons.loadLambda(it.url!!),
+                // We can runBlocking here to get the icon since we are already on an IO thread
+                icon = runBlocking { icons?.loadIcon(IconRequest(it.url!!))?.await()?.bitmap },
```

Comment:
I like this solution better than blocking inside `IconLoader` as it's not blocking the main thread. This is blocking the provider thread though until the icon is loaded, meaning that we won't get any suggestions from this provider if an icon is slow to load. So, a slow loading icon will delay displaying *all* suggestions from this provider. This is mitigated by our icon cache and we could add a timeout (upper bound) when fetching icons and fall back to a default icon to address this. It seems like a path forward (make sure the suggestion is ready, has an icon, before it's being displayed), but wondering what @pocmo thinks about it :).

Reply:
@sblatz however, since this is only called from a `suspend` function, we can remove the `runBlocking` by either moving the `loadIcon` out of `into`, or simply making into `suspend` as well. This will have the same consequence of only creating the suggestion once we have the icon.